src/mem/alloc.h

```diff
@@ -469,16 +629,20 @@ namespace snmalloc
 #endif
     }
 
+    /*
+     * Free memory of an unknown size. Must be called with an external
+     * pointer.
+     */
     SNMALLOC_FAST_PATH void dealloc(void* p)
     {
 #ifdef USE_MALLOC
       return free(p);
 #else
 
-      // Free memory of an unknown size. Must be called with an external
-      // pointer.
       uint8_t size = pagemap().get(address_cast(p));
 
+      p = pagemap().getp(p);
```

Comment:
This is on the fast path? Ensure it is erased on non-Cheri case. Perhaps add a comment?